app/src/main/java/com/example/weatherandroidapp/views/WeatherGridFragment.kt

```diff
@@ -20,7 +26,24 @@ class WeatherGridFragment : Fragment() {
         savedInstanceState: Bundle?
     ): View? {
         binding = WeatherGridFragmentBinding.inflate(layoutInflater)
-        // TODO("implement view model")
         return binding.root
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.mutableMainState.observe(viewLifecycleOwner, Observer { updateUI(it) })
+        viewModel.getWeatherInfo()
+    }
+
+    private fun updateUI(data: Event<Data<ResponseWeatherInfo>>) {
+        when (data.peekContent().status) {
+            Status.UPLOADED_DATA -> {
+                Toast.makeText(this.context, getString(R.string.toast_show_ok_service, data.peekContent().data?.city?.name, data.peekContent().data?.list?.get(0)?.mainResponse?.temperature.toString()), Toast.LENGTH_SHORT).show()
+            }
+            Status.ERROR -> {
+                Toast.makeText(this.context, data.peekContent().error.toString(), Toast.LENGTH_LONG).show()
+            }
+        }
```

Comment:
this can be inside with(data.peekContent()){ ... }

Reply:
Done